src/data_access/SearchDataAccessObject.java

```diff
@@ -69,38 +65,58 @@ public SearchReponseArray getData(SearchRequest searchRequest) {
     try (Response response = client.newCall(request).execute()) {
 
       JSONObject rootNode = new JSONObject(response.body().string());
+      System.out.println(query.toString());
+      System.out.println(rootNode.toString());
       JSONArray hitsArray = rootNode.getJSONObject("hits").getJSONArray("hits");
       ArrayList<SearchResponse> searchResponses = new ArrayList<>();
-      Pattern pattern = Pattern.compile("<em>(.*?)</em>");
       for (int i = 0; i < hitsArray.length(); i++) {
         JSONObject hit = hitsArray.getJSONObject(i);
         JSONObject source = hit.getJSONObject("_source");
         JSONArray highlightNode = hit.getJSONObject("highlight").optJSONArray("message");
         String finalHighlight;
         if (highlightNode != null) {
-          ArrayList<String> object = new ArrayList<>();
-          for (int j = 0; j < highlightNode.length(); j++) {
-            object.add(highlightNode.getString(j));
-          }
-          String highlightedText = object.get(0);
-          Matcher matcher = pattern.matcher(highlightedText);
-          matcher.find();
-          finalHighlight = matcher.group().substring(4, 8);
+          finalHighlight = highlightNode.getString(0);
         } else {
           finalHighlight = "";
         }
+
+        String openTag = "<em>";
+        String closeTag = "</em>";
+        List<SearchIndicies> item = new ArrayList<>();
+        String text = finalHighlight;
+        int currentIndex = 0;
+        while (text.indexOf(openTag, currentIndex) != -1) {
+          int start = text.indexOf(openTag, currentIndex) + openTag.length();
+          int end = text.indexOf(closeTag, start);
+
+          if (end != -1) {
+            item.add(
+                new SearchIndicies(
+                    start - (openTag.length() * item.size() + closeTag.length() * item.size()),
+                    end
+                        - (openTag.length() * (item.size() + 1)
```

Comment:
I'm confused what this code is doing exactly, but I'll trust that it's correct. Make sure to double check please

Reply:
Ok, I'll comment the code to explain this, as a comment there